src/main/java/com/fakng/fakngagrgtr/parser/apple/AppleParser.java

```diff
@@ -59,8 +65,50 @@ protected List<Vacancy> getAllVacancies() throws Exception {
         return allVacancies;
     }
 
-    private HtmlPage getPage(int numberOfPage) throws IOException {
-        return downloadPage(String.format(url, numberOfPage));
+    @Override
+    protected Iterable<Vacancy> enrichVacanciesWithDetails(Iterable<Vacancy> vacancies) throws IOException {
+        for (Vacancy vacancy : vacancies) {
+            if (!vacancy.isFullyConstructed()) {
+                HtmlPage vacancyPage = downloadPage(vacancy.getUrl());
+                vacancy.setDescription(buildDescription(vacancyPage));
+                vacancy.setFullyConstructed(true);
+                try {
+                    Thread.sleep(1000);
```

Comment:
This is added in order not to abuse apple site. We can think about a better pausing mechanism between requests.